lib/Doctrine/ORM/PersistentCollection.php

```diff
@@ -240,11 +240,12 @@ public function getSnapshot()
      */
     public function getDeleteDiff()
     {
-        return array_udiff_assoc(
-            $this->snapshot,
-            $this->collection->toArray(),
-            function($a, $b) { return $a === $b ? 0 : 1; }
-        );
+        $collectionItems = $this->collection->toArray();
+
+        return \array_values(\array_diff_key(
+            \array_combine(\array_map('spl_object_hash', $this->snapshot), $this->snapshot),
```

Comment:
Note for later, maybe we should use `spl_object_id` as mentionned here: https://github.com/doctrine/orm/pull/6878

Reply:
It's in the backport todo list, yes :)